src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,67 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+
+type Props = {
+  todo: Todo;
+  onClose: () => void;
+};
+
+export const TodoModal: React.FC<Props> = ({ todo, onClose }) => {
+  const [user, setUser] = useState<User | null>(null);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    getUser(todo.userId)
+      .then(setUser)
+      .finally(() => setIsLoading(false));
```

Comment:
The `getUser` function should handle errors in case the API call fails. Consider adding a `.catch` block to handle potential errors and set an appropriate state to inform the user.